core/src/main/scala/org/apache/spark/util/MetadataCleaner.scala

```diff
@@ -76,7 +76,7 @@ private[spark] object MetadataCleanerType extends Enumeration {
 // initialization of StreamingContext. It's okay for users trying to configure stuff themselves.
 private[spark] object MetadataCleaner {
   def getDelaySeconds(conf: SparkConf): Int = {
-    conf.getInt("spark.cleaner.ttl", -1)
+    Utils.timeStringAsS(conf.get("spark.cleaner.ttl", "-1s")).toInt
```

Comment:
I can just make it -1 as it was before. The function will assume it's a seconds.